addons/mail_plugin/models/res_partner.py

```diff
@@ -5,4 +5,39 @@
 class ResPartner(models.Model):
     _inherit = 'res.partner'
 
-    iap_enrich_info = fields.Text('IAP Enrich Info', help='Stores additional info retrieved from IAP in JSON')
+    iap_enrich_info = fields.Text('IAP Enrich Info', help='IAP response stored as a JSON string',
```

Comment:
Why not adding an iap_search_domain doing the same kind of computation ? It would allow creating partners with iap_enrich_info and iap_search_domain and hide complexity within this technical model.

Side issue to check: giving both of them at the same time, we should not create two lines ^^

Reply:
@tde-banana-odoo It work :) (it do not create 2 res.partner.iap), but it's more efficient to override the create / write in that case no ? (to perform only 1 search and not 2)

Reply:
We could override create when having both given if we want to speedup process indeed. This will probably be the most common use case.

Reply:
Or even better, I make a computed field `partner_iap_id` and computed method depends on it :)